### ISSUE TITLE
Cannot change location of local recordings

### ISSUE THREAD

**Issue by gncs** [flathub/us.zoom.Zoom#257]
I cannot edit the location of local recordings. By default, it's set to `/home/<user>/Documents/Zoom`, and I cannot change that. With the system-wide installation provided by Zoom (not the flatpak), this is possible. Is this the intended behavior? I understand that the application is not supposed to have access to my entire home directory, but it should be able to save to `~/.var/app/us.zoom.Zoom/*`, for example, no?

**Comment by vchernin** [comment]
From a brief look it seems like a classic sandbox file system issue. Zoom seems to only be able to look at directories it is given explict access too, and when I try to change it doesn't seem to use the file chooser portal dialog. So either we give zoom more access to the file system, or we get zoom to use the portal to save files.

As a workaround you could use [flatseal](https://flathub.org/apps/details/com.github.tchx84.Flatseal) and add more file system access to Zoom manually.

Is there an easy way to force QT apps to use the file chooser portal?

**Comment by toupeira** [comment]
@vchernin in my case the portal is already used, if I install `xdg-desktop-portal-gtk` on Debian and use Zoom from Flathub. But the problem seems to be that it's requesting a file rather than a directory, so selecting a directory will just open it, and selecting a file will close the dialog and do nothing.

I haven't tried the KDE portal yet (it pulls in a lot of dependencies...), but I wonder if it has the same bug or lets you select and save a directory path.

Some related discussion here: https://forums.linuxmint.com/viewtopic.php?t=335395

**Comment by vchernin** [comment]
@toupeira What version of `xdg-desktop-portal` and `xdg-desktop-portal-gtk` do you have? Only somewhat recent versions of the file chooser portal support directories, and there were many fixes and improvements to that feature since then.

**Comment by toupeira** [comment]
@vchernin I'm on Debian unstable with these versions:

```
ii  xdg-desktop-portal         1.8.1-1      amd64        desktop integration portal for Flatpak and Snap
ii  xdg-desktop-portal-gtk     1.8.0-1+b1   amd64        GTK+/GNOME portal backend for xdg-desktop-portal
```

**Comment by vchernin** [comment]
> But the problem seems to be that it's requesting a file rather than a directory, so selecting a directory will just open it, and selecting a file will close the dialog and do nothing.

Yeah upon trying it out locally on Fedora 34 Silverblue it seems you're correct. The portal is used but used incorrectly. Maybe it's being called incorrectly. Or maybe there's a bug in [xdg-desktop-portal](https://github.com/flatpak/xdg-desktop-portal) or QT. 

I wonder if this occurs with other QT apps trying to choose a directory with the portal?

**Comment by toupeira** [comment]
Had a quick look and found https://flathub.org/apps/details/org.cubocore.CoreArchiver, I can choose directories there correctly! But it's using the KDE framework, and Zoom is using a bundled Qt I think, so this probably doesn't tell us much :grinning: 

I also tried out https://flathub.org/apps/details/org.flatpak.qtdemo but it only has a demo for file selection.

**Comment by vchernin** [comment]
> Zoom is using a bundled Qt I think, so this probably doesn't tell us much 

True, although it might work a bit differently [in this branch](https://github.com/flathub/us.zoom.Zoom/pull/255) where the KDE runtime is used (but still probably bundled QT).

**Comment by vchernin** [comment]
Same issues happens in the Wayland branch.. this is most likely due to Zoom using Qt 5.12 still.

**Comment by zazke** [comment]
I'll just add that I did manage to change the location, but after trying to record a meeting it just failed to save the file.

After reading [this reply by grx_pattix on Linux Mint Forum](https://forums.linuxmint.com/viewtopic.php?t=335395)

> I've found the right config file.
> If you follow the path "/var/lib/flatpak/app/us.zoom.Zoom/x86_64/stable/very_long_directory_name/" should be a file called "metadata", inside it there is the line
> "filesystems=~/Documents/Zoom:create;~/.zoom:create;", I tryed to change the path and everything worked :)

I did the following:

1. Change the `~/Documents/Zoom:create` to `~/desired/path/dir`
2. Restart Zoom flatpak
3. Remove the directory in `~/desired/path/dir` and instead created an empty file
4. Selected the file using Zoom GUI
5. Removed the empty file and replaced it with a directory again
6. Started and recorded a test meeting.  Tried to save the recording but failed

I guess this issue sadly remains open after about year and a half. :disappointed:

**Comment by valiantgenomics** [comment]
This issue still persists on Zoom 5.12.6.173 (flatpak). I cannot change my recordings folder location.

**Comment by oryband** [comment]
I just lost all zoom recordings I made today, Zoom is silently failing to save recordings.
